core/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -211,68 +215,81 @@ private SearchHit createSearchHit(SearchContext context, FieldsVisitor fieldsVis
         return searchHit;
     }
 
-    private SearchHit createNestedSearchHit(SearchContext context, int nestedTopDocId, int nestedSubDocId, int rootSubDocId, Set<String> fieldNames, List<String> fieldNamePatterns, LeafReaderContext subReaderContext) throws IOException {
+    private SearchHit createNestedSearchHit(SearchContext context, int nestedTopDocId, int nestedSubDocId,
```

Comment:
Note: other changes outside this method was to remove the checkstyle suppression.